utils/src/utils/validationUtils.ts

```diff
@@ -0,0 +1,31 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+
+export namespace validationUtils {
+    export interface RangeConstraints {
+        lowerLimitIncl?: number;
+        upperLimitIncl?: number;
+    }
+
+    export function hasValidCharLength(value: string, rc?: RangeConstraints): boolean {
+        const lowerLimitIncl = (!rc?.lowerLimitIncl || rc.lowerLimitIncl < 1) ? 1 : rc.lowerLimitIncl;
+        const upperLimitIncl = (!rc?.upperLimitIncl || rc.upperLimitIncl > Number.MAX_SAFE_INTEGER) ? Number.MAX_SAFE_INTEGER : rc.upperLimitIncl;
+        return lowerLimitIncl <= upperLimitIncl && value.length >= lowerLimitIncl && value.length <= upperLimitIncl;
+    }
+
+    export function getInvalidCharLengthMessage(rc?: RangeConstraints): string {
```

Comment:
Could you explain how this is working? I'm not really sure how you could figure out which message to output without having the `value.length` as part of the function.

I'm not understanding how the `lowerLimitIncl` existing but not the `upperLimitIncl` can be extrapolated to mean that the value needs to be greater than the lower limit.

Oh actually, I think I understand now. The message isn't reactive to the value-- so if the error is too few characters, it will still output `The value must be between {0} and {1} characters long.'` if both limits exist.

I don't know about other people, but my assumption made this very confusing to me, but I'm not sure if my expectation that it would output a message relevant to the error is realistic.

Also-- I feel like the lower limit should default to 1. I can't think of any case where we wouldn't have at least a requirement of 1 character.

Reply:
Actually disregard this:

> I don't know about other people, but my assumption made this very confusing to me, but I'm not sure if my expectation that it would output a message relevant to the error is realistic.

It's easier just to maintain the messages with how it is now. I think it's a pretty common error message pattern anyway.

Reply:
> Could you explain how this is working? I'm not really sure how you could figure out which message to output without having the value.length as part of the function.

In this pattern, you would need to determine whether the message needs to be output before calling getMessage, then you pass it your constraints, which defines what the error message needs to say.  The error message wording in this case is always constant, and tied to the constraints which are known ahead of time.  

> I'm not understanding how the lowerLimitIncl existing but not the upperLimitIncl can be extrapolated to mean that the value needs to be greater than the lower limit.

To me a boundary not existing means there is not limit in that direction.  I hope that's a safe assumption, I feel like it is for this.

I'm thinking of it like: 
[lowerLimit, upperLimit]: lowerLimit <= x <= upperLimit (same as `x >= lowerLimit` && `x <= upperLimit`)
[2, 5]: 2 <= x <= 5 (same as `x >= 2 && x <= 5`)
[5, _]: 5 <= x (same as `x >= 5`)

> I don't know about other people, but my assumption made this very confusing to me, but I'm not sure if my expectation that it would output a message relevant to the error is realistic.

I based it around the common pattern I observed in the extensions, which was something like:

```
if (!valid) {
  return "not valid for x reason";
}
```

We could always change this pattern, there's other validation libraries which we could take inspiration from instead.  

> Also-- I feel like the lower limit should default to 1. I can't think of any case where we wouldn't have at least a requirement of 1 character.

For the actual validation logic, it does default to 1.  For the message portion, it _kind of_ basically does this, but is only explicit about the number in the message if you supply the exact value.  I guess the only case that I think might look weird is if you put in a lower limit of `1`.  I think it would look better to show the first message in that case, but that's just a preference and not sure how much we care about that edge case (let me know if you have a preference)